Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -957,29 +957,18 @@ public AndConstraint<TAssertions> ContainInOrder(params T[] expected)
         /// </param>
         public AndConstraint<TAssertions> ContainItemsAssignableTo<TExpectation>(string because = "", params object[] becauseArgs)
         {
-            bool success = Execute.Assertion
+            Execute.Assertion
                 .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:collection} to contain at least one element assignable to type {0}{reason}, ",
+                    typeof(TExpectation).FullName)
                 .ForCondition(Subject is not null)
-                .FailWith("Expected {context:collection} to contain element assignable to type {0}{reason}, but found <null>.",
-                    typeof(TExpectation));
-
-            if (success)
-            {
-                int index = 0;
-                foreach (T item in Subject)
-                {
-                    if (item is not TExpectation)
-                    {
-                        Execute.Assertion
-                            .BecauseOf(because, becauseArgs)
-                            .FailWith(
-                                "Expected {context:collection} to contain only items of type {0}{reason}" +
-                                ", but item {1} at index {2} is of type {3}.", typeof(TExpectation), item, index, item.GetType());
-                    }
-
-                    ++index;
-                }
-            }
+                .FailWith("but found <null>.")
+                .Then
+                .Given(() => Subject.ConvertOrCastToCollection())
+                .ForCondition(subject => subject.Any(x => typeof(TExpectation).IsAssignableFrom(GetType(x))))
+                .FailWith("but found {0}.", subject => subject.Select(x => x.GetType()))
```

Comment:
I think you'll need to replace `x.GetType()` with `GetType(x)`. 
Otherwise the failure message of 
```cs
new []{ typeof(int) }.Should().ContainItemsAssignableTo<string>()
```
would contain `RuntimeType` instead of `System.Int32` as we call `typeof(int32).GetType()`.

Reply:
Done in https://github.com/fluentassertions/fluentassertions/pull/1765/commits/65a30b1aa34d735173589b02dc188542ac3f6b51.